### ISSUE TITLE
cfnlint via stdin ignored when templates config defined in .cfnlintrc

### ISSUE THREAD

**Issue by halfmu** [aws-cloudformation/cfn-lint#4135]
### CloudFormation Lint Version

1.34.2

### What operating system are you using?

Mac

### Describe the bug

When `templates` config is not present in `.cfnlintrc`, stdin works as expected:
```sh
echo 'Resources: {}' | cfn-lint
```

When `templates` config is present in `.cfnlintrc`, stdin is ignored, which causes issues for linting plugins via stdin in IDE:
```sh
echo -e 'templates:\n  - abc.yaml' > .cfnlintrc
echo 'Resources: {}' | cfn-lint
E0003 abc.yaml could not be processed by glob.glob
None:1:1
```

### Expected behavior

`stdin` should take precedence before `templates` config in `.cfnlintrc`:

```sh
echo -e 'templates:\n  - abc.yaml' > .cfnlintrc
echo 'Resources: {}' | cfn-lint
```

### Reproduction template

```yaml
Resources: {}
```

```yaml
# .cfnlintrc
templates:
   - abc.yaml
```

**Comment by kddejong** [comment]
Because of this line: https://github.com/aws-cloudformation/cfn-lint/blob/bd0c0ce80d581281cf14e0d93099f5c1a5f50b5b/src/cfnlint/runner.py#L399

We can't just switch it to `if not sys.stdin.isatty():` as templates passed via a parameter should still take precedence.  The best way to do this would be to validate if we are getting templates from a .cfnlintrc file or from parameters.

**Comment by mbarneyjr** [comment]
I think this is still an issue:

```
$ ls -al
$ cfn-lint --version
cfn-lint 1.43.2
$ echo 'Resources: {}' | cfn-lint
$ echo -e 'templates:\n  - abc.yaml' > .cfnlintrc
$ echo 'Resources: {}' | cfn-lint
E0003 abc.yaml could not be processed by glob.glob
None:1:1
```
